packages/vkui/src/hooks/useTodayDate.ts

```diff
@@ -11,23 +12,55 @@ import { getMillisecondsToTomorrow } from '../lib/date';
  * @param listenDayChangesForUpdate - флаг по которому определяется, будет ли создаваться подписка на смену календарного дня
  */
 export function useTodayDate(listenDayChangesForUpdate = false) {
+  const { document } = useDOM();
   const [todayDate, setTodayDate] = React.useState(() => new Date());
 
   React.useEffect(() => {
-    if (!listenDayChangesForUpdate) {
+    if (!listenDayChangesForUpdate || !document) {
       return;
     }
 
-    const timeToDayChange = getMillisecondsToTomorrow(todayDate);
+    let timeout: NodeJS.Timeout | null = null;
```

Comment:
clearTimeout может принимать `undefined`

```suggestion
    let timeout: NodeJS.Timeout | undefined = undefined;
```

Reply:
Вообще, правильней указать

```ts
let timeout: number | undefined = undefined;
```

а `clearTimeout` использовать через `window.clearTimeout`, тогда TS поймёт, что речь идёт о браузерной имплементации.